src/main/java/org/threadly/examples/features/PrioritySchedulerExample.java

```diff
@@ -0,0 +1,164 @@
+   * handles the args on startup
+   */
+  private static Boolean handleArgs(final String[] args) {
+    //handles the args, decides whether to call help and validates data in args
+    try {
+      Integer.parseInt(args[0]);
+    } catch (NumberFormatException e) {
+      printHelp();
+      return false;
+    } catch (IndexOutOfBoundsException e) {
+      printHelp();
+      return false;
+    }
+    try {
+      if (args[1].equalsIgnoreCase("-v")) {
+        PrioritySchedulerExample.threadDetail = true;
+      }
+    } catch (IndexOutOfBoundsException e) {
+      PrioritySchedulerExample.threadDetail = false;
+    }
+    return true;
+  }
+  
+  /**
+   * Prints the information
+   */
+  public static void printHelp() {
+    System.out.println("Priority Scheduler Example executes a number of threads with a known, but randomly assigned. Each thread counts to the Max Integer Value before exiting.");
+    System.out.println("args: numThreads -> the integer numner of threads to be executed");
+    System.out.println("optional args: \"-v\" -> increases the amount of data reported by each thread and reduced the threads to counting to 100");
+    System.out.println("ex: PrioritySchedulerExample 1000 -v");
+  }
+  
+  /**
+   * gets a pseudorandom TaskPriority value
+   */
+  private static TaskPriority getRandomPriority() {
+    Random rand = new Random();
+    int randNum = rand.nextInt((2) + 1);
+    if (randNum == 0) {
+      return TaskPriority.Starvable;
+    } else if (randNum == 1) {
+      return TaskPriority.Low;
+    } else {
+      return TaskPriority.High;
+    }
+  }
+  
+  
+  /**
+   * This is a generic Runnable that the PriorityScheduler can run.
+   * 
+   */
+  private static class BasicRunnable implements Runnable {
+    
+    /**
+     * the id of the thread which will be reported to stdout
+     */
+    private final int runNumber;
+    
+    /**
+     * the start time of the runnable
+     */
+    private final double startTime;
+    
+    /**
+     * whether or not to display detail
+     */
+    private final Boolean detail;
+    
+    /**
+     * Parameterized ctor. note: parameters are not required for a runnable to be executed by the PriorityScheduler.
+     * @param pRunNumber -> the id number of the thread which thread will report to stdout
+     * @param pDetail -> whether or not to show verbose output (suppress for large number of threads for readability)
+     */
+    public BasicRunnable(final int pRunNumber, final Boolean pDetail) {
+      this.runNumber = pRunNumber;
+      this.startTime = System.nanoTime();
+      this.detail = pDetail;
+    }
+    
+    @Override
+    public void run() {
+      for (int i = 0; i < (detail ? 100 : Integer.MAX_VALUE); i++) {
+        if (detail) {
+          System.out.println("Thread " + runNumber + " is on iteration " + (i) + ". ");
+        }
+      }
+      System.out.println("Thread " + runNumber + " took " + (System.nanoTime() - startTime) + "  nanoseconds to execute." );
```

Comment:
Instead of System.nanoTime, check out the Clock class: http://threadly.github.io/threadly/javadocs/4.3.0/org/threadly/util/Clock.html

We should have another example class for that, so no need to go deeply into it, but we might as well eat our own dog food in our examples.